app/models/group_assignment_repo.rb

```diff
@@ -28,8 +28,10 @@ class GroupAssignmentRepo < ApplicationRecord
   before_validation(on: :create) do
     if organization
       create_github_repository
-      push_starter_code
-      add_team_to_github_repository
+      delete_github_repository_on_failure do
+        push_starter_code
+        add_team_to_github_repository
+      end
```

Comment:
This diff is the major change. To understand what this does look at the definition of  [`#delete_github_repository_on_failure`](https://github.com/education/classroom/blob/c358de13a4cc526daee9a5ec58d845c6d4e8564a/app/models/concerns/github_repoable.rb#L33-L40).
https://github.com/education/classroom/blob/c358de13a4cc526daee9a5ec58d845c6d4e8564a/app/models/concerns/github_repoable.rb#L33-L40